ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/utils/ThreadUtil.java

```diff
@@ -50,18 +53,28 @@ public static String getThreadId()
      */
     public static String getThreadId(final Thread thread)
     {
-	final Object id = THREAD_ID.get() ;
+	final Object id = THREAD_ID.get(thread) ;
```

Comment:
This is a String so use final String id = THREAD_ID.get(thread) ; and then you don't need to do the cast on the next line.